pkg/doc/signature/suite/bbsblssignatureproof2020/signer.go

```diff
@@ -45,29 +45,51 @@ func (s *Suite) SelectiveDisclosure(doc map[string]interface{}, revealDoc map[st
 
 	delete(docCompacted, "proof")
 
-	proofInfo, err := getBlsProof(rawProofs)
+	blsSignatures, err := getBlsProofs(rawProofs)
 	if err != nil {
 		return nil, err
 	}
 
-	blsProof := proofInfo.blsProof
+	var revealDocumentResult map[string]interface{}
+
+	for i, blsProof := range blsSignatures {
+		revealedDoc, derivedProof, dErr := generateSignatureProof(blsProof, resolver, docCompacted,
+			revealDoc, nonce, opts...)
+		if dErr != nil {
+			return nil, dErr
+		}
+
+		if i == 0 {
+			revealDocumentResult = revealedDoc
+			revealDocumentResult["proof"] = make([]map[string]interface{}, len(blsSignatures))
+		}
+
+		proofs := revealDocumentResult["proof"].([]map[string]interface{}) //nolint:errcheck
```

Comment:
why `nolint:errcheck` here?

Reply:
This code is not clean. I will simplify this.

Reply:
Done.